### PR TITLE
Find command Closes #46

### LINKED ISSUE #46: Find Expense

### PR DESCRIPTION
**Pull request by jinbesan**

### CODE REVIEW

**Review by nicleejy — APPROVED** [review]
Find command implementation and test cases look good. Merge approved.

File: data/addressbook.json [hunk 1]
```diff
@@ -1,70 +1,31 @@
...
+    "categoryName" : "food",
+    "summary" : "summary"
+  } ],
+  "expenses" : [ {
+    "name" : "apple",
+    "amount" : "1.5",
+    "date" : "2023-03-16",
+    "category" : {
+      "categoryName" : "food",
+      "summary" : "summary"
```
Comment (gitsac) [inline data/addressbook.json:12]: Nice to see that the sample data is well done!

**Review by gitsac — APPROVED** [review]
Good sample data and test cases, LGTM!

**Comment by shirsho-12** [comment]
closes #46

File: docs/UserGuide.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ layout: page
 title: FastTrack User Guide
 ---
 
-FastTrack is an easy-to-use **financial management desktop application** designed for NUS undergraduate students who are living on a tight budget. 
+FastTrack is an easy-to-use **financial management desktop application** designed for NUS undergraduate students who are living on a tight budget.
```
Comment (shirsho-12) [inline docs/UserGuide.md:6]: Good improvements to DG

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+public class FindCommand extends Command {
+
+    public static final String COMMAND_WORD = "find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all expense which names contain any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " apple orange cherry";
+
+    private final ExpenseContainsKeywordsPredicate predicate;
```
Comment (shirsho-12) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:22]: Code follows codebase style

File: src/test/java/seedu/address/model/expense/ExpenseContainsKeywordsPredicateTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different person -> returns false
+        assertFalse(firstPredicate.equals(secondPredicate));
+    }
```
Comment (shirsho-12) [inline src/test/java/seedu/address/model/expense/ExpenseContainsKeywordsPredicateTest.java:43]: Good testing principles.

**Review by shirsho-12 — APPROVED** [review]
LGTM

**Pull request merged**